api/script/index.js

```diff
@@ -564,68 +564,89 @@ module.exports = function transferFlow({utError: {fetchErrors}}) {
+                    };
+                });
+            });
 
-                    return result;
-                })
-                .catch(error => {
-                    params.abortAcquirer = error;
-                    return this.bus.importMethod('transferFlow.push.execute')(params, $meta);
-                })
-                .then(result => Object.assign(params, {
-                    cardProductName: result.cardProductName,
-                    sourceAccount: result.sourceAccountNumber,
-                    sourceAccountName: result.sourceAccountName,
-                    destinationAccount: result.destinationAccountNumber,
-                    sourceCardProductId: result.sourceCardProductId,
-                    destinationAccountName: result.destinationAccountName,
-                    issuerId: result.issuerId,
-                    ledgerId: result.ledgerId,
-                    cardNumber: result.cardNumber,
-                    ordererId: result.ordererId
-                }))
-                .then(result => !params.transferIdAcquirer && this.bus.importMethod(`db/${params.channelType}.terminal.nextId`)({
-                    channelId: result.channelId
-                }, {forward}))
-                .then(result => {
-                    if (params.transferIdAcquirer) {
-                        return params;
-                    }
-                    if (!result || !result[0] || !result[0][0] || !result[0][0].tsn) {
-                        throw errors['transfer.nextId']();
-                    }
-                    params.transferIdAcquirer = result[0][0].tsn;
-                    return params;
-                })
-                .then(params => this.bus.importMethod('transferFlow.push.execute')(params, $meta));
+            const processAcquirerError = Promise.resolve().then(() => {
+                if (!params.abortAcquirer) {
+                    return {};
+                }
+                return this.bus.importMethod('db/atm.card.processError')({
+                    cardId: params.cardId,
+                    errorType: params.abortAcquirer.type
+                }, $meta);
+            });
+
+            const catchError = promise => promise.catch(error => {
+                this.log && this.log.error && this.log.error(error);
+                return {abortAcquirer: error};
+            });
+
+            return Promise.all([
+                catchError(acquirerId),
+                catchError(cardDetails),
+                catchError(processAcquirerError)
+            ]).then(([
+                acquirerId,
+                cardDetails,
+                processAcquirerError
+            ]) => {
+                const transfer = Object.assign({}, acquirerId, cardDetails, params);
+                const selectFirstError = errors => errors.find(e => e);
+                transfer.abortAcquirer = selectFirstError([
+                    processAcquirerError.abortAcquirer,
```

Comment:
Isn't processAcquirerError an array?

Reply:
the procedure error is catch just like before the changes